test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -0,0 +1,2081 @@
+		);
+	}
+);
+
+// Add load operation to unset statement arguments recursively
+static YPR<Statement> addLoadToStmtRec(
+	[](Statement* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Statement>(
+			[](Statement* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::addArgsRec(
+					_message,
+					[](Expression* _expr, YulRandomNumGenerator& _rand)
+					{
+						auto tmp = YPM::loadExpression(_rand);
+						_expr->CopyFrom(*tmp);
+						delete tmp;
+					},
+					_rand
+				);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Make statement arguments load expression recursively"
+		);
+	}
+);
+
+// Add load from zero location ops to unset statement arguments recursively
+static YPR<Statement> addLoadFromZeroToStmtRec(
+	[](Statement* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Statement>(
+			[](Statement* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::addArgsRec(
+					_message,
+					[](Expression* _expr, YulRandomNumGenerator& _rand)
+					{
+						auto tmp = YPM::loadFromZero(_rand);
+						_expr->CopyFrom(*tmp);
+						delete tmp;
+					},
+					_rand
+				);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Make statement arguments loads from location zero recursively"
+		);
+	}
+);
+
+// Add binop expression to statement.
+static YPR<Statement> addBinopToStmt(
+	[](Statement* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Statement>(
+			[](Statement* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::addArgs(_message, YPM::binopExpression, _rand);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Make statement arguments binary operations"
+		);
+	}
+);
+
+// Mutate varref
+static YPR<VarRef> mutateVarRef(
+	[](VarRef* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<VarRef>(
+			[](VarRef* _message, YulRandomNumGenerator& _rand)
+			{
+				_message->set_varnum(_rand());
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Mutate variable reference"
+		);
+	}
+);
+
+// Add load expression to statement
+static YPR<Statement> addLoadToStmt(
+	[](Statement* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Statement>(
+			[](Statement* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::addArgs(_message, YPM::loadExpression, _rand);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Mutate statement arguments to a load expression"
+		);
+	}
+);
+
+// Add a randomly chosen statement to a statement block
+static YPR<Block> addStmt(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::addStmt(_message, _rand);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add pseudo randomly chosen statement type to statement block"
+		);
+	}
+);
+
+void YPM::addArgs(
+	Statement *_stmt,
+	std::function<Expression *(YulRandomNumGenerator&)> _func,
+	YulRandomNumGenerator& _rand
+)
+{
+	switch (_stmt->stmt_oneof_case())
```

Comment:
Most of the tests below have the form `!something || something.expr().expr_oneof_case()`. Could that be simplified somehow? Maybe a function inside `something` that already checks both?

Reply:
Good point, I'll refactor them into a function.